integration/src/main/java/scripts/testscripts/ListWorkspaces.java

```diff
@@ -0,0 +1,93 @@
+package scripts.testscripts;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+import bio.terra.testrunner.runner.config.TestUserSpecification;
+import bio.terra.workspace.api.WorkspaceApi;
+import bio.terra.workspace.model.WorkspaceDescription;
+import bio.terra.workspace.model.WorkspaceDescriptionList;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import scripts.utils.ClientTestUtils;
+import scripts.utils.WorkspaceAllocateTestScriptBase;
+
+/**
+ * This test exercises WSM's listWorkspaces endpoint. Note that because this test reads the full
+ * set of a user's workspaces, it is not thread safe and only one instance of this test should
+ * run against a server at once.
+ */
+public class ListWorkspaces extends WorkspaceAllocateTestScriptBase {
+
+  private UUID workspaceId2;
+  private UUID workspaceId3;
+  private WorkspaceApi secondUserApi;
+
+  @Override
+  protected void doSetup(List<TestUserSpecification> testUsers, WorkspaceApi workspaceApi) throws Exception {
+    assertThat("There must be two test users to run this test",
+        testUsers.size() == 2);
+    // Primary test user (index 0) is handled in the test base.
+    TestUserSpecification nonWorkspaceUser = testUsers.get(1);
+    secondUserApi = ClientTestUtils.getWorkspaceClient(nonWorkspaceUser, server);
+    // We expect test users to never have permanent workspaces. However, they may have some
+    // workspaces due to previous test failures. Before running this test, we clean up all
+    // workspaces for both test users.
+    WorkspaceDescriptionList preTestWorkspaceList = workspaceApi.listWorkspaces(/*offset=*/0, /*limit=*/10000);
+    for (WorkspaceDescription workspace : preTestWorkspaceList.getWorkspaces()) {
+      workspaceApi.deleteWorkspace(workspace.getId());
+    }
+    WorkspaceDescriptionList secondUserPreTestWorkspaceList = secondUserApi.listWorkspaces(/*offset=*/0, /*limit=*/10000);
+    for (WorkspaceDescription workspace : secondUserPreTestWorkspaceList.getWorkspaces()) {
+      secondUserApi.deleteWorkspace(workspace.getId());
```

Comment:
I don't know if we have a use case outside of this test, but a batch endpoint would help here.

Reply:
I'm not sure when else we would use that either. Maybe for folders, but that's a pretty underdesigned area.